tests/conftest.py

```diff
@@ -1902,42 +1903,59 @@ def rgw_bucket_factory_session(request, rgw_obj_session):
 
 
 @pytest.fixture()
-def bucket_factory(request, mcg_obj):
+def bucket_factory(request, bucket_class_factory, mcg_obj):
     """
     Returns an MCG bucket factory
     """
-    return bucket_factory_fixture(request, mcg_obj)
+    return bucket_factory_fixture(request, bucket_class_factory, mcg_obj)
 
 
 @pytest.fixture(scope="session")
-def bucket_factory_session(request, mcg_obj_session):
+def bucket_factory_session(request, bucket_class_factory_session, mcg_obj_session):
     """
     Returns a session-scoped MCG bucket factory
     """
-    return bucket_factory_fixture(request, mcg_obj_session)
+    return bucket_factory_fixture(
+        request, bucket_class_factory_session, mcg_obj_session
+    )
 
 
-def bucket_factory_fixture(request, mcg_obj=None, rgw_obj=None):
+def bucket_factory_fixture(
+    request, bucket_class_factory=None, mcg_obj=None, rgw_obj=None
+):
     """
     Create a bucket factory. Calling this fixture creates a new bucket(s).
     For a custom amount, provide the 'amount' parameter.
 
     Args:
+        bucket_class_factory: creates a new Bucket Class
         mcg_obj (MCG): An MCG object containing the MCG S3 connection
             credentials
         rgw_obj (RGW): An RGW object
 
     """
     created_buckets = []
 
-    def _create_buckets(amount=1, interface="S3", verify_health=True, *args, **kwargs):
+    def _create_buckets(
+        amount=1,
+        interface="S3",
+        verify_health=True,
+        bucketclass: typing.Union[None, str, dict] = None,
```

Comment:
Why is there explicitly specified type by `typing` module? It is not done in any other part of our framework.

Reply:
It's meant as an additional calrifying layer regarding the current state of the variable.
More on the reason it can be one of three options [in this comment](https://github.com/red-hat-storage/ocs-ci/pull/3437#discussion_r539238628)

Reply:
Var was changed to be either `dict` or `None`, and thus `typing` was ommitted. Resolving conversation.